php-frontend/src/main/java/org/sonar/php/tree/symbols/SymbolVisitor.java

```diff
@@ -0,0 +1,372 @@
+        if (symbol != null) {
+          symbol.addUsage(tree);
+        }
+      }
+    }
+
+    classMemberUsageState = null;
+  }
+
+  private boolean isBuiltInVariable(VariableIdentifierTree tree) {
+    return BUILT_IN_VARIABLES.contains(tree.text().toUpperCase());
+  }
+
+  @Override
+  public void visitCompoundVariable(CompoundVariableTree tree) {
+    SyntaxToken firstExpressionToken = ((PHPTree) tree.variableExpression()).getFirstToken();
+    if (!firstExpressionToken.text().startsWith("$")) {
+      Symbol symbol = currentScope.getSymbol("$" + firstExpressionToken.text());
+      if (symbol != null) {
+        symbol.addUsage(firstExpressionToken);
+      }
+    }
+
+    super.visitCompoundVariable(tree);
+  }
+
+  @Override
+  public void visitParameter(ParameterTree tree) {
+    createSymbol(tree.variableIdentifier(), Symbol.Kind.PARAMETER);
+    // do not scan the children to not pass through variableIdentifier
+  }
+
+  @Override
+  public void visitGlobalStatement(GlobalStatementTree tree) {
+    for (VariableTree variable : tree.variables()) {
+      // Other cases are not supported
+      if (variable.is(Tree.Kind.VARIABLE_IDENTIFIER)) {
+        IdentifierTree identifier = (IdentifierTree) variable.variableExpression();
+        Symbol symbol = globalScope.getSymbol(identifier.text(), Symbol.Kind.VARIABLE);
+        if (symbol != null) {
+          currentScope.addSymbol(symbol);
+        } else {
+          createSymbol(identifier, Symbol.Kind.VARIABLE);
+        }
+
+      }
+    }
+  }
+
+  @Override
+  public void visitLexicalVariables(LexicalVariablesTree tree) {
+    for (VariableTree variable : tree.variables()) {
+      IdentifierTree identifier = null;
+      if (variable.is(Tree.Kind.VARIABLE_IDENTIFIER)) {
+        identifier = (IdentifierTree) variable.variableExpression();
+      } else if (variable.is(Tree.Kind.REFERENCE_VARIABLE) && variable.variableExpression().is(Tree.Kind.VARIABLE_IDENTIFIER)) {
+        identifier = ((VariableIdentifierTree) variable.variableExpression()).variableExpression();
+      }
+
+      if (identifier != null) {
+        Symbol symbol = currentScope.outer().getSymbol(identifier.text());
+        if (symbol != null) {
+          symbol.addUsage(identifier);
+        } else if (variable.is(Kind.REFERENCE_VARIABLE)) {
+          symbolTable.declareSymbol(identifier, Symbol.Kind.VARIABLE, currentScope.outer());
+        }
+        createSymbol(identifier, Symbol.Kind.VARIABLE);
+      }
+    }
+
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    if (tree.callee().is(Tree.Kind.NAMESPACE_NAME)) {
+      NamespaceNameTree namespaceNameCallee = (NamespaceNameTree) tree.callee();
+      usageForNamespaceName(namespaceNameCallee, Symbol.Kind.FUNCTION);
+
+    }
+
+    this.insideCallee.push(true);
+    tree.callee().accept(this);
+    this.insideCallee.pop();
+
+    // fixme optimize through visitor
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)